Sources/Valet/SecureEnclaveValet.swift

```diff
@@ -226,13 +246,78 @@ public final class SecureEnclaveValet: NSObject {
 
     // MARK: Internal Properties
 
+    /// A provider of an `LAContext` used when needing to manually evaluate an authentication policy
+    /// prior to retrieving a value from the keychain in order to allow the evaluation prompt to show a
+    /// custom fallback button to the user.  Should always return a new instance of `LAContext` in
+    /// practice, but specified as an internal var to allow it to be overridden in tests.
+    internal var authenticationContextProvider: () -> LAContext = { LAContext() }
     internal let service: Service
 
     // MARK: Private Properties
 
     private let lock = NSLock()
     private let baseKeychainQuery: [String : AnyHashable]
 
+    // MARK: - Private Methods
+
+    /// Synchronously calls `LAContext.evaluatePolicy`, passing the evaluated context
+    /// via `kSecUseAuthenticationContext` when querying for the string at the given
+    /// key in the secure enclave.
+    private func synchronouslyEvaluatePolicy(
+        forKey key: String,
+        withPrompt userPrompt: String,
+        withFallbackTitle fallbackTitle: String
+    ) throws -> String {
+        // Use a semaphore to block the thread and perform the evaluation synchronously.
+        let semaphore = DispatchSemaphore(value: 0)
+        var result: Result<String, Error>?
+
+        let authenticationContext = authenticationContextProvider()
+        authenticationContext.localizedFallbackTitle = fallbackTitle
+        authenticationContext.evaluatePolicy(
+            accessControl.policy,
+            localizedReason: userPrompt,
+            reply: { [unowned self] success, error in
+                if success {
+                    do {
+                        var keychainQuery = baseKeychainQuery
+                        keychainQuery[kSecUseAuthenticationContext as String] = authenticationContext
+                        let string = try SecureEnclave.string(
+                            forKey: key,
+                            withPrompt: userPrompt,
+                            options: keychainQuery
+                        )
+                        result = .success(string)
+
+                    } catch {
+                        result = .failure(error)
+                    }
+                } else if let error = error {
+                    result = .failure(error)
+
+                } else {
+                    // Unexpected to get here
+                    result = .failure(KeychainError.couldNotAccessKeychain)
```

Comment:
This doesn't feel like the right error to use. It sucks that Apple's API is using a separate `success` bool and `error` property (I guess that's just Objective-C for us) which is forcing us to come up with an error case that should never exist. But we should probably create a new error (not a `KeychainError`) for this case. 

I wonder if we could create an `enum FallbackError` with a case for the customer selecting the fallback and another for this `internalError` case? (Open to naming revisions as well as alternative ideas!)

Reply:
Yeah I didn't love using this error here either. I wonder if we should even try to be handling this? The documentation says that `success` will indicate success or failure and if `success == false`, the error will be populated. Sucks to have to put those assumptions in manually and not have it explicit though

Reply:
Given that we're in Swift I think we _have_ to handle it, right? Otherwise we'd have to `fatalError` or something if we don't have either a retrieved value or error to propagate. And crashing because Apple screwed up feels like a poor option.